src/ngraph/graph_util.cpp

```diff
@@ -179,6 +180,45 @@ void ngraph::replace_node(std::shared_ptr<Node> target, std::shared_ptr<Node> re
     target->clear_control_dependents();
 }
 
+size_t
+    ngraph::replace_by_friendly_name(const shared_ptr<Function>& f,
+                                     const unordered_map<string, shared_ptr<Node>>& replacement_map)
+{
+    size_t num_replacements = 0;
+
+    for (auto& replacee : f->get_ops())
```

Comment:
Could you also add a check that graph contains only one node for each name from the unordered_map?